### ISSUE TITLE
file.managed creating zero-sized files from HTTP source

### ISSUE THREAD

**Issue by jerrykan** [saltstack/salt#44410]
### Description of Issue/Question
There seems to be a bug in the latest release of salt where files that are sourced from HTTP are not correctly fetched, but instead are just created as zero-sized files.

Using a git bisect it seems the problem commit is b838460816

### Setup
With the follow SLS:
```
gitea-package-deps:
  pkg.installed:
    - name: git

gitea-binary-dir:
  file.directory:
    - name: /opt/gitea

gitea-binary:
  file.managed:
    - name: /opt/gitea/gitea
    - source: http://<remote host>/gitea/gitea-1.2.1-linux-amd64
    - source_hash: sha256=83b894e321343e253006e689fb5a4d6fee048cf0fa3a169cb54c4060788133df
    - mode: 755
    - require:
      - file: gitea-binary-dir
```
I am getting a error about mismatched hashes.

```
local:
----------
          ID: gitea-package-deps
    Function: pkg.installed
        Name: git
      Result: True
     Comment: All specified packages are already installed
     Started: 16:33:42.686547
    Duration: 187.468 ms
     Changes:   
----------
          ID: gitea-binary-dir
    Function: file.directory
        Name: /opt/gitea
      Result: True
     Comment: Directory /opt/gitea is in the correct state
              Directory /opt/gitea updated
     Started: 16:33:42.876388
    Duration: 0.485 ms
     Changes:   
----------
          ID: gitea-binary
    Function: file.managed
        Name: /opt/gitea/gitea
      Result: False
     Comment: Specified sha256 checksum for /opt/gitea/gitea (83b894e321343e253006e689fb5a4d6fee048cf0fa3a169cb54c4060788133df) does not match actual checksum (e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855)
     Started: 16:33:42.877114
    Duration: 461.837 ms
     Changes:   
              ----------
              diff:
                  New file

Summary for local
------------
Succeeded: 2 (changed=1)
Failed:    1
------------
Total states run:     3
Total run time: 649.790 ms
```

If I replace the `source_hash` with `skip_verify` the `/opt/gitea/gitea` file is created but has a file size of 0 

### Steps to Reproduce Issue
The following is the the wireshark capture of Salt downloading the files (before apparently throwing it away):

Request:
```
GET /gitea/gitea-1.2.1-linux-amd64 HTTP/1.1
Connection: close
Host: <remote host>
User-Agent: Salt/2017.7.0-121-gb838460 http.query()
```
Response
```
HTTP/1.1 200 OK
Date: Mon, 06 Nov 2017 05:13:08 GMT
Server: Apache/2.4.10 (Debian)
Last-Modified: Thu, 19 Oct 2017 03:47:10 GMT
ETag: "3128198-55bde3608af92"
Accept-Ranges: bytes
Content-Length: 51544472
Connection: close

... (contents of the binary file) ...
```

### Versions Report
```
salt-call --versions-report
Salt Version:
           Salt: 2017.7.2
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Jan 19 2017, 14:48:08)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1
 
System Versions:
           dist: debian 9.1 
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-3-amd64
         system: Linux
        version: debian 9.1 
```

**Comment by Ch3LL** [comment]
This should fix it for ya: https://github.com/saltstack/salt/pull/44016 which will be included in the 7.3 release

**Comment by jerrykan** [comment]
@Ch3LL Yes, I can confirm this is fixed in the current develop branch. Apologies, I should have tested against develop before filing the issue and wasting your time.
